### ISSUE TITLE
Request and Response Transformer: values with comma

### ISSUE THREAD

**Issue by thefosk** [Mashape/kong#1295]
When adding a new header in the Request or Response transformer with a comma, like:

```
X-Hello: 1234,67
```

Kong only adds:

```
X-Hello: 1234
```

**Comment by Tieske** [comment]
is this with, or without #1289 ?

**Comment by thefosk** [comment]
I think it's unrelated - and caused by how we are parsing arrays. Probably this line here: https://github.com/Mashape/kong/blob/master/kong/dao/schemas_validation.lua#L103

**Comment by Tieske** [comment]
Cannot reproduce this on 0.8.3;

```
http POST localhost:8001/apis name=mockbin request_path=/headers upstream_url=http://mockbin.org
http POST localhost:8001/apis/mockbin/plugins name=request-transformer config.add.headers[1]=hi:there,and,here
http GET http://localhost:8000/headers
```

results in;
```
thijs$ http GET http://localhost:8000/headers
HTTP/1.1 200 OK
Access-Control-Allow-Credentials: true
Access-Control-Allow-Headers: host,connection,accept-encoding,x-forwarded-for,cf-ray,x-forwarded-proto,cf-visitor,accept,user-agent,hi,cf-connecting-ip,x-request-id,x-forwarded-port,via,connect-time,x-request-start,total-route-time
Access-Control-Allow-Methods: GET
Access-Control-Allow-Origin: *
CF-RAY: 2c1f8f0ffcbd51a0-SJC
Connection: keep-alive
Content-Encoding: gzip
Content-Type: application/json; charset=utf-8
Date: Wed, 13 Jul 2016 20:41:59 GMT
Etag: W/"4f6-Gb+SBQfWN8IvffyIRLgHqQ"
Server: cloudflare-nginx
Set-Cookie: __cfduid=d61b1b088f451b5d15b8906f9afaa5e651468442519; expires=Thu, 13-Jul-17 20:41:59 GMT; path=/; domain=.mockbin.org; HttpOnly
Transfer-Encoding: chunked
Vary: Accept, Accept-Encoding
Via: kong/0.8.3
X-Kong-Proxy-Latency: 0
X-Kong-Upstream-Latency: 205
X-Powered-By: mockbin

{
    "headers": [
        {
            "name": "host",
            "value": "mockbin.org"
        },
        {
            "name": "connection",
            "value": "close"
        },
        {
            "name": "accept-encoding",
            "value": "gzip"
        },
        {
            "name": "x-forwarded-for",
            "value": "127.0.0.1,40.140.33.170, 162.158.255.237"
        },
        {
            "name": "cf-ray",
            "value": "2c1f8f0ffcbd51a0-SJC"
        },
        {
            "name": "x-forwarded-proto",
            "value": "http"
        },
        {
            "name": "cf-visitor",
            "value": "{\"scheme\":\"http\"}"
        },
        {
            "name": "accept",
            "value": "*/*"
        },
        {
            "name": "user-agent",
            "value": "HTTPie/0.9.3"
        },
        {
            "name": "hi",
            "value": "there,and,here"            <======== here it is
        },
        {
            "name": "cf-connecting-ip",
            "value": "40.140.33.170"
        },
        {
            "name": "x-request-id",
            "value": "e22f8be8-6a32-4899-82a6-5bade7660dff"
        },
        {
            "name": "x-forwarded-port",
            "value": "80"
        },
        {
            "name": "via",
            "value": "1.1 vegur"
        },
        {
            "name": "connect-time",
            "value": "0"
        },
        {
            "name": "x-request-start",
            "value": "1468442519078"
        },
        {
            "name": "total-route-time",
            "value": "0"
        }
    ],
    "headersSize": 499
}
```

**Comment by Tieske** [comment]
Issue only manifests itself when using items, in a comma-separated way, without array notation. This is not supported and should be done using the array notation.

This is the correct way;
```
http POST localhost:8001/apis/mockbin/plugins name=request-transformer "config.add.headers[1]=hi:there,and,here" "config.add.headers[2]=hi2:and,here"
```

This is accepted, but doesn't work;
```
http POST localhost:8001/apis/mockbin/plugins name=request-transformer config.add.headers=hi:there,and,here
```

The latter results in this config;
```
        "add": {
            "body": {},
            "headers": [
                "hi:there",
                "and",
                "here",
                ""
            ],
            "querystring": {}
        },
```

**Comment by Tieske** [comment]
fixed in #1399
